// prettier/server.js
function __args() {
  var setting = {};
  //判断参数个数是否为1并且是否为字符串
  if (arguments.length === 1 && typeof arguments[0] !== 'string') {
    //如果是那么为url
    setting = arguments[0];
  } else {
    //如果不是的话分为两种情况
    //参数的第一项肯定为url
    setting.url = arguments[0];
    //判断参数的第二项是否为对象
    if (typeof arguments[1] === 'object') {
      //如果是那么第二项为传的参数
      arguments[1].agentid = 0;
      arguments[1].storeid = 34;
      setting.data = arguments[1];

      //第三项为成功方法
      setting.success = arguments[2];
    } else {
      //如果不是   那么第二项为成功方法
      setting.success = arguments[1];
    }
  }
  //如果没有拼接前面的httpxxx给他拼接上
  if (setting.url.indexOf('https://') !== 0) {
    setting.url = getApp().globalData.localhost + setting.url;
  }
  console.log('请求地址为'+setting.url);
  //返回
  return setting;
}
//接收方法和参数
function __json(method, setting) {
  //拼接了setting内的参数
  setting.method = method;
  setting.header = {
    'content-type': 'application/json'
  };
  uni.request(setting);//修改这里
}
function __pjson(method, setting) {
  setting.method = method;
  setting.header = {
    'content-type': 'application/json'
  };
  uni.request(setting);//修改这里
}
function __image(method, setting) {
  setting.method = method;
  setting.header = {
    "Content-Type": "multipart/form-data"
  };
  uni.request(setting);
}

//使其他地方可以引用
export default {
  getJSON: function () {
    __json('GET', __args.apply(this, arguments));
  },
  postJSON: function () {
    __pjson('POST', __args.apply(this, arguments));
  },
  postIMAGE: function () {
    __image('POST', __args.apply(this, arguments));
  },
  sendTemplate: function (formId, templateData, success, fail) {
    var app = getApp();
    this.getJSON({
      url: '/WxAppApi/sendTemplate',
      data: {
        rd_session: app.rd_session,
        form_id: formId,
        data: templateData,
      },
      success: success,   // errorcode==0时发送成功
      fail: fail
    });
  }
}
